### PR TITLE
feat: allow to add new roles and avoid restart after upsert an user

### PR DESCRIPTION
**Pull request by alvindaiyan**
## Description

allow to add new roles and avoid restart after upsert an user

Screenshot 2023-10-31 at 15 32 33

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

### CODE REVIEW

**Comment by NingLu** [comment]
We identified new issues on unchanged lines of code. Navigate to the Amazon CodeGuru Reviewer console to view the recommendations to fix them.

File: aws_extension/cloud_api_manager/api_manager.py [hunk 1]
```diff
@@ -194,6 +194,24 @@ def list_roles(self, user_token=""):
         raw_resp.raise_for_status()
         return raw_resp.json()
 
+    def upsert_role(self, role_name, permissions, creator, user_token=""):
```
Comment (NingLu) [inline aws_extension/cloud_api_manager/api_manager.py:197]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Possible hardcoded password: &#39;&#39; https://bandit.readthedocs.io/en/latest/plugins/b107_hardcoded_password_default.html

File: aws_extension/cloud_api_manager/api_manager.py [hunk 2]
```diff
@@ -194,6 +194,24 @@
...
+
+        raw_resp = requests.post(f'{cloud_auth_manager.api_url}role', json=payload, headers=self._get_headers_by_user(user_token))
+        raw_resp.raise_for_status()
+        resp = raw_resp.json()
+        if resp['statusCode'] != 200:
+            raise Exception(resp['errMsg'])
+
+        return True
+
     def upsert_user(self, username, password, roles, creator, initial=False, user_token=""):
```
Comment (NingLu) [inline aws_extension/cloud_api_manager/api_manager.py:215]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Possible hardcoded password: &#39;&#39; https://bandit.readthedocs.io/en/latest/plugins/b107_hardcoded_password_default.html

File: aws_extension/cloud_api_manager/api_manager.py [hunk 3]
```diff
@@ -194,6 +194,24 @@
...
+            "role_name": role_name,
+            "permissions": permissions,
+            "creator": creator
+        }
+
+        raw_resp = requests.post(f'{cloud_auth_manager.api_url}role', json=payload, headers=self._get_headers_by_user(user_token))
+        raw_resp.raise_for_status()
+        resp = raw_resp.json()
+        if resp['statusCode'] != 200:
+            raise Exception(resp['errMsg'])
```
Comment (NingLu) [inline aws_extension/cloud_api_manager/api_manager.py:211]: Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Using `Exception` and `BaseException` can make your code prone to errors and difficult to maintain. Instead, we recommend using one of the [Built-in Exceptions](https://docs.python.org/3/library/exceptions.html) or creating a custom exception class that is derived from `Exception` or one of its subclasses.

**Review by elonniu — APPROVED** [review]

**Pull request merged**